### PR TITLE
Apr 2566 unsuccessful gw pass oversight overturned

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/SFA.DAS.ApplyService.Web/Services/OverallOutcomeService.cs [hunk 1]
```diff
@@ -2,6 +2,7 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+using NPOI.SS.Formula.Functions;
```
Comment (martin-lush) [inline src/SFA.DAS.ApplyService.Web/Services/OverallOutcomeService.cs:5]: I don't think this should be here

File: src/SFA.DAS.ApplyService.Web/Services/OverallOutcomeService.cs [hunk 2]
```diff
@@ -97,38 +98,48 @@ public ApplicationSummaryViewModel BuildApplicationSummaryViewModel(Apply applic
...
-                    applicationUnsuccessfulModerationFail = true;
+                    if (oversightReview.ModerationApproved.HasValue && oversightReview.ModerationApproved == true)
+                    {
+                        moderationFailedAndApproved = true;
+                    }
+
+                    if (oversightReview.ModerationApproved.HasValue && oversightReview.ModerationApproved == false)
+                    {
+                        moderationFailedAndOverturned = true;
+                    }
```
Comment (martin-lush) [inline src/SFA.DAS.ApplyService.Web/Services/OverallOutcomeService.cs:124]: This took me a few moments to figure out what was happening. How about this to cover both cases:

`if (oversightReview.ModerationApproved.HasValue)
 {
        moderationFailedAndApproved = oversightReview.ModerationApproved;
 }`

**Review by martin-lush — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.ApplyService.Web/Services/OverallOutcomeService.cs (hunk 2 above)
Comment (martin-lush) [inline src/SFA.DAS.ApplyService.Web/Services/OverallOutcomeService.cs:124]: Took a few moments to see what was going on here. Personally I would make it a:  _if - else if_ statement

File: src/SFA.DAS.ApplyService.Web/Services/OverallOutcomeService.cs [hunk 3]
```diff
@@ -97,38 +98,48 @@ public ApplicationSummaryViewModel BuildApplicationSummaryViewModel(Apply applic
...
+            var moderationPassedAndApproved = false;
+            var gatewayPassOverturnedToFail = false;
+            var moderationFailedAndOverturned = false;
             if (application?.GatewayReviewStatus == GatewayAnswerStatus.Pass)
             {
-                if (application?.ModerationStatus != null
-                    && application?.ModerationStatus == ModerationStatus.Fail
-                    && oversightReview.ModerationApproved.HasValue
-                    && oversightReview.ModerationApproved == true)
+                if (application?.ModerationStatus != null && application?.ModerationStatus == ModerationStatus.Fail)
```
Comment (martin-lush) [inline src/SFA.DAS.ApplyService.Web/Services/OverallOutcomeService.cs:114]: can remove `application?.ModerationStatus != null` as the null coalesce will do that

File: src/SFA.DAS.ApplyService.Web/Services/OverallOutcomeService.cs [hunk 4]
```diff
@@ -97,38 +98,48 @@ public ApplicationSummaryViewModel BuildApplicationSummaryViewModel(Apply applic
...
+                        moderationFailedAndApproved = true;
+                    }
+
+                    if (oversightReview.ModerationApproved.HasValue && oversightReview.ModerationApproved == false)
+                    {
+                        moderationFailedAndOverturned = true;
+                    }
                 }
 
                 if (application?.ModerationStatus != null
```
Comment (martin-lush) [inline src/SFA.DAS.ApplyService.Web/Services/OverallOutcomeService.cs:127]: same here as above can remove `application?.ModerationStatus != null` as the null coalesce will do that

**Review by martin-lush — APPROVED** [review]

File: src/SFA.DAS.ApplyService.Web/Services/OverallOutcomeService.cs (hunk 1 above)
Comment (MarkFCain) [reply src/SFA.DAS.ApplyService.Web/Services/OverallOutcomeService.cs:5]: will remove

**Pull request merged**